packages/axios-extension/src/auth/uaa.ts

```diff
@@ -257,4 +299,14 @@ export class Uaa {
         this.log.info('Got access token successfully');
         return response?.data?.access_token;
     }
+
+    /**
+     * @returns an access token using the BTP UAA credentials
+     */
+    public async getAccessTokenUsingClientCredentials(): Promise<string> {
+        const tokenRequest = this.getTokenRequestForClientCredential();
+        const response: AxiosResponse = await axios.request(tokenRequest);
```

Comment:
What are the failure modes for this? How should they be handled?

Reply:
When a client requires access without having to interact with a user but the application requires a user context for granting access, the `password` grant type is used. This requires sharing the user credentials with the client id and secret. Thus this works the same as the `getAccessToken` flow where the grant type is `refresh_token` if a refresh token is already passed in, or `authorization_code` grant type if a new refresh_tokn is required.
Not sure if that answers your question?

Reply:
Thanks for the explanation. My question is more about handling any failures in the code

Reply:
@longieirl  I think Ullas is questioning about there is no try-catch error handling for the axios request. `getAccessTokenWithClientCredentials` is called in `attachUaaAuthInterceptor`, which also doesn't handle errors.

Reply:
Thanks for clarifying that Shen. That's exactly what I was trying to say as well

Reply:
@ullasholla are you referring to there being no try/catch? It's the calling client that needs to handle this since its the interceptor that is setting up the appropiate headers for all the subsequent API calls to work. For example, we make a GET and then a POST or a PUT. On the initial GET, if this fails with the UAA properties provided when using either `getAccessToken` or `getAccessTokenWithClientCredentials`, the its deploy-tooling to decide how to proceed, either prompt for credentials or ignore and exit the process.

Reply:
Is the error handling in `handleAxiosError()`?

Do we try to prompt for creds for an HTTP 401?
<img width="796" alt="image" src="https://github.com/SAP/open-ux-tools/assets/212316/ec897849-31db-4467-bb5c-79c188547f0e">

If yes, in the pipeline, this will just sit there with the prompt and then time out

Reply:
No, the `handleError` in ui5-tooling is configured with the following code;
```
if (!config.noRetry && isAxiosError(error)) {
```
So for a ci/cd pipepline flow, it will exit the process and log the error.

Reply:
Quickly tested it out with the `deploy` command instead of `undeploy`. `config.noRetry` was undefined:
![image](https://github.com/SAP/open-ux-tools/assets/212316/11e4d4f4-8fff-4593-a436-e4a9add1aaa6)

I forced the status code to be 401 and I get this:
![image](https://github.com/SAP/open-ux-tools/assets/212316/ef8e5747-05f6-492c-9434-bbed29c10128)

Did not exit with an error for me.

Could you quickly test it once for me please? Please test `deploy` like I did

Reply:
@ullasholla good catch! No idea how this was working before! So there is a bug with the commander module, using the param `--no-retry` does not map to `noRetry`. I had to change it to `--abort-retry`. From locally testing with Command v9, two letter words dont work as expected!

```
./bin/deploy --name ZJLONG0205 --package '$tmp' --transport '' --url 'https://abcd.ondemand.com' --cloud true --cloud-service-env  --safe -n
2023-06-14 22:05:23 info    abap-deploy-task: Creating archive from dist. 
2023-06-14 22:05:23 info    abap-deploy-task: Archive created from dist. 
2023-06-14 22:05:23 warn    abap-deploy-task: You chose not to validate SSL certificate. Please verify the server certificate is trustful before proceeding. See documentation for recommended configuration (https://help.sap.com/viewer/17d50220bcd848aa854c9c182d65b699/Latest/en-US/4b318bede7eb4021a8be385c46c74045.html). 
2023-06-14 22:05:23 info    abap-deploy-task: Starting to deploy. 
2023-06-14 22:05:23 error   abap-deploy-task: Deployment has failed. 
2023-06-14 22:05:23 error   abap-deploy-task: Change logging level to debug your issue
        (see examples https://github.com/SAP/open-ux-tools/tree/main/packages/deploy-tooling#configuration-with-logging-enabled) 
Request failed with status code 401
```
So now, the `-n` and  `--abort-retry` work as expected.

FYI @tobiasqueck

Reply:
Passing `-n` or `--abort-retry` works fine now, thanks @longieirl.

For the particular usecase we're discussing - using deploy/undeploy in a pipeline and having trouble with credentials - the user needs to explicitly pass `-n/--abort-retry`

This is because we cannot assume using `--cloud-service-env` implies that it's being used in a CI env.

In our doco it's not clear that not retrying is the same as suppressing prompts.
This is what we have:
`  -n, --abort-retry                    do not retry if deploy fails for any reason (default: false)`

John, @theStiggySpd, @ianquigley-sap  & @tobiasqueck, I can think of a few things we can do to make this clearer to the user (not all mutually exclusive):
* mention it in the `--abort-retry` doco
* add a 'non-interactive' option to do the same thing
* add an example for the CI scenario in the doco

Thoughts?

Reply:
@ullasholla I agree, I've added another task to the story to update documentation around this flow.
As regards the non-interactive mode, I will add that as an additional task to the original story as its currently only being supported by ui5-tooling for the moment.

Reply:
Further documentation to be appended.